app/models/miq_report/generator/trend.rb

```diff
@@ -24,33 +24,24 @@ def build_results_for_report_trend(options)
     #   :trend_filter   => MiqExpression#object
     #   :trend_db       => VmPerformance
     # }
+    trend_klass = db_options[:trend_db]
+    trend_klass = Object.const_get(trend_klass) unless trend_klass.kind_of?(Class)
     if db_options[:interval] == "daily"
-      results = []
-
       includes = include.blank? ? [] : include.keys
+      timestamp = Metric::Helper.timestamp_from_offset(db_options[:start_offset], db_options[:end_offset], tz)
 
-      start_time, end_time = Metric::Helper.get_time_range_from_offset(db_options[:start_offset], db_options[:end_offset], :tz => tz)
-      trend_klass = db_options[:trend_db].kind_of?(Class) ? db_options[:trend_db] : Object.const_get(db_options[:trend_db])
-
-      recs = VimPerformanceDaily.find_entries(:class => trend_klass, :tz => tz, :time_profile => time_profile)
-             .where(where_clause).where(:timestamp => start_time..end_time).includes(includes)
-      results = Rbac.filtered(recs, :class        => db_options[:trend_db],
-                                    :filter       => db_options[:trend_filter],
-                                    :userid       => options[:userid],
-                                    :miq_group_id => options[:miq_group_id])
+      recs = Metric::Helper.find_for_interval_name("daily", trend_klass, time_profile, tz)
+                           .where(where_clause).where(:timestamp => timestamp).includes(includes)
     else
       start_time = Time.now.utc - db_options[:start_offset].seconds
```

Comment:
@kbrock I believe that the difference in logic has to do with whether the report interval is daily vs. hourly. For daily reports we want to have the start_time be at the beginning of the day in the specified time zone so the calculation starts at the end of the previous day (so that the first hour of the day would be the first selected). For an hourly report, the day does not matter so it will just subtract the offset from the current time UTC.